src/GraphQL/SchemaExtensions.cs

```diff
@@ -1,13 +1,12 @@
 using System;
 using System.Threading.Tasks;
-using GraphQL.Http;
 using GraphQL.Types;
 
 namespace GraphQL
 {
     public static class SchemaExtensions
     {
-        public static async Task<string> ExecuteAsync(this ISchema schema, Action<ExecutionOptions> configure)
+        public static async Task<string> ExecuteAsync(this ISchema schema, IDocumentWriter documentWriter, Action<ExecutionOptions> configure)
```

Comment:
I'm feeling like adding the `IDocumentWriter` parameter here pretty much defeats the purpose of the method.  I don't like it, but is seems like this method would have to move to the `NewtonsoftJson` lib.

That's making me think that the abstraction is wrong here.  Seems like we're moving too much to the other library.

Reply:
There is a second SchemaExtensions class in GraphQL.NewtonsoftJson.

    public static class SchemaExtensions
    {
```
        public static Task<string> ExecuteAsync(this ISchema schema, Action<ExecutionOptions> configure)
        {
            return schema.ExecuteAsync(new DocumentWriter(indent: true), configure);
        }
    }
```

This allows a user to still have the simpler extension if they've chosen to use NewtonJson.

The only way to keep this entirely in GraphQL would be to make the result be a `Task<ExecutionResult>` instead of a `Task<String>`

Reply:
No, do not worry. We transferred as much as needed. In such situations, boundary places always appear, where additional configuration is required. Previously, they were simply invisible, since this entire configuration was hard-coded inside.

Reply:
👍 Missed the other half of that, thanks.  Yeah I'm thinking that return result from the executor should be revisited at some point.